Support Debian Trixie in `powershell` Molecule scenario, fix bug in `csharp` Molecule scenario

jsf9k opened the pull request:
## 🗣 Description ##

This pull request:
- Adds a workaround to support Debian Trixie in this Ansible role's `powershell` Molecule scenario
- Fixes a bug that was causing the `csharp` Molecule scenario tests to fail for Kali

## 💭 Motivation and context ##

- See #71.  These changes are also needed for cisagov/teamserver-packer#107 and cisagov/egress-assess-packer#53.
- This was a bug that I noticed while implementing the Debian Trixie workaround.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.
- [x] Bump major, minor, patch, pre-release, and/or build versions [as appropriate](https://semver.org/#semantic-versioning-specification-semver) via the `bump_version` script *if* this repository is versioned *and* the changes in this PR [warrant a version bump](https://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api).

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release (necessary if and only if the version was bumped).

jsf9k committed:
Add workaround for outdated dependency of PowerShell/PowerShell

Debian Trixie has libicu76, but the deb package generated for
PowerShell/PowerShell only supports libicu74 for now.  This is the
case even though PowerShell/PowerShell functions fine with libicu76.
See PowerShell/PowerShell#25865 for more information.

Until a new release is created, we use Debian's equivs package to
install a dummy dependency for libicu74, which is sufficient to
install PowerShell/PowerShell.  When run, PowerShell/PowerShell will
use the libicu76 dependency installed from the official Debian Trixie
package repository.

jsf9k committed:
Bump version from 2.0.1 to 2.0.2

jsf9k committed:
Install mono-devel vs mono-complete for Kali

When one installs mono-complete on Kali, apt now actually installs
mono-devel.

jsf9k committed:
Prefer pytest.fail() to assert False for unconditional failures

mcdonnnj commented on tasks/install_powershell_Debian.yml:
```diff
@@ -1,4 +1,45 @@
+# install PowerShell/PowerShell.  When run, PowerShell/PowerShell will
+# use the libicu76 dependency installed from the official Debian
+# Trixie package repository.
+- name: Build and install equivs dummy package for libicu74 for Trixie
```
I guess my one question here is how does this ensure that libicu76 is installed and available?

jsf9k replied:
`libicu` is pretty fundamental and hence is always around, but I created commit 593a3c0 just in case.

On non-Trixie platforms a version of `libicu` commensurate with the latest PowerShell release will be pulled in as a dependency of PowerShell, so Trixie is the only special case we need worry about.

mcdonnnj replied:
Yep I figured we were generally ok, but since the dependency would no longer be pulled in automatically I just wanted to be safe.

jsf9k replied:
It was a good catch.

jsf9k committed:
Also install libicu76 on Debian Trixie

This package has always just been there, but by explicitly installing
it we ensure that is the case.  Otherwise, PowerShell could get
installed on Trixie with only the libicu74 dummy/equivs package.

The ICU library is fairly fundamental, which probably explains why it
was always "just there":
https://unicode-org.github.io/icu/userguide/icu/

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>
